packager/packager.cc

```diff
@@ -1227,16 +1228,22 @@ class ImportTreeBuilder final {
                 // but effectively use the one from the export site (due to the export-mapping/public module).
                 // This results in the following behavior:
                 // imported constant: `Foo::Bar::Baz` from package `Foo::Bar`
-                //                     ^^^^^^^^       jump to package file of `Foo::Bar`
+                //                     ^^^^^^^^       jump to top of package file of `Foo::Bar`
                 //                               ^^^  jump to actual definition of `Baz` class
 
                 // Ensure import's do not add duplicate loc-s in the test_module
-                const bool useImportLoc = source.isEnumeratedImport &&
-                                          (moduleType != ModuleType::PrivateTest || source.isFriendOrTestImport());
-                const auto &importLoc = useImportLoc ? source.importLoc : core::LocOffsets::none();
+                const auto &moduleLoc = [](auto &source, auto packageLoc) {
+                    // normal or test import
+                    if (source.isTestImport() || source.isNormalImport()) {
+                        return source.isEnumeratedImport ? source.importLoc : core::LocOffsets::none();
+                    }
+
+                    // friend import
+                    return packageLoc->offsets();
+                }(source, packageLoc);
```

Comment:
It's pleasing that the local variable here can be declared `const`, but the immediately-evaluated lambda strikes me as non-idiomatic C++.

Reply:
Can move it to a helper function.

Reply:
++

Reply:
Fixed